spec/lib/api/v3/work_packages/work_package_representer_spec.rb

```diff
@@ -507,6 +505,39 @@
         end
       end
 
+      context 'when the user does not have the permission to remove watchers' do
+        before do
+          role.permissions.delete(:delete_work_package_watchers) and role.save
+        end
+
+        it 'should not have a link to add watcher' do
+          expect(subject).not_to have_json_path('_links/removeWatcher/href')
+        end
+      end
+
+      describe 'watchers link' do
+        context 'when the user is allowed to see watchers' do
+          it_behaves_like 'has an untitled link' do
+            let(:link) { 'watchers' }
+            let(:href) { api_v3_paths.work_package_watchers work_package.id }
+          end
+
+          it 'embeds the watchers as collection' do
+            is_expected.to be_json_eql('Collection'.to_json).at_path('_embedded/watchers/_type')
```

Comment:
We have that already, however I don't want to test the representer for user collection here. I only want to verify that the wp representer utilizes such a representer and then there is another test checking whether the user collection representer does sane things.

Now I only need an approximation for "the wp representer uses a user collection representer in apropriate places". The approximation I've chosen here (and I think in other places too) is that there should be the type `Collection` at a specific path.

An evil developer could make this test green by re-implementing the user collection representer inline in the wp representer. But evil developers are hard to detect via unit tests ;-)